src/login/pam_systemd.c

```diff
@@ -211,21 +211,39 @@ static int export_legacy_dbus_address(
         return r;
 }
 
+static int append_session_limit(sd_bus_message *m, const char *limit, const char *name) {
+        uint64_t val;
+        int r;
+
+        if (!isempty(limit)) {
+                r = parse_size(limit, 1024, &val);
+                /* Don't return here on an invalid value, as that would simply deny the login */
+                if (r >= 0) {
+                        r = sd_bus_message_append(m, "(sv)", name, "t", val);
+                        if (r < 0)
+                                return r;
+                }
+        }
```

Comment:
i am pretty sure there should be a way to unset these limits too. see how this is done in bus_append_cgroup_property() in src/shared/bus-unit-util.c. (that file contains the parsers for the command line settings you pass to "systemctl set-property" and such). i.e. "infinity" and things should be understood...

Moreover I have the suspicion that sooner or later people also want to be able to specify relative settings here, like "systemctl set-property" allows too, see the same function